### PR TITLE
ITB backfill container (SOFTWARE-5002)

### PR DESCRIPTION
**Pull request by matyasselmeci**
https://opensciencegrid.atlassian.net/browse/SOFTWARE-5002

If you specify `--build-arg ITB=1` when building the container, it will use the itb versions of the osg-flock scripts and also add the attributes for joining the ITB pool.

Other features (intended for manual builds):
- specify your fork of osg-flock in the `OSG_FLOCK_REPO` var, e.g. `--build-arg OSG_FLOCK_REPO=matyasselmeci/osg-flock`
- specify the branch of osg-flock in the `OSG_FLOCK_BRANCH` var, e.g. `--build-arg OSG_FLOCK_BRANCH=mybranch`
- the above two args, as well as `OSG_FLOCK_HASH` (the git hash of the checkout) will be advertised in the startd ad
- specify `--build-arg RANDOM=$RANDOM` to use the docker build cache for the RPM install, but get a fresh checkout of the scripts. This saves a lot of time, especially on slow connections.

The -development-itb tag gets built in the workflow; I don't build -release-itb and -testing-itb because the build matrix is already pretty big.

Includes #60.

### CODE REVIEW

**Review by rynge — COMMENTED** [review]
Should it maybe also register to the ITB pool? I don't have strong feelings about this as you can also control access via the START expression.

**Comment by matyasselmeci** [comment]
Sure, what do I need to do for that?

**Comment by matyasselmeci** [comment]
Is that what pilots do? Does that mean you'll have to do `condor_status -pool cm-1.ospool-itb.osg-htc.org` to see them?

**Comment by matyasselmeci** [comment]
OK, I changed it to use the ITB CMs by default, but you can turn it off by setting `ITBPOOL=0` in the environment.

File: Dockerfile [hunk 1]
```diff
@@ -125,6 +125,8 @@ RUN if [[ -n $TIMESTAMP ]]; then \
 
 RUN \
     if [[ -n $ITB ]]; then \
+        # Set the default pool to ITB, but allow turning off with -e ITBPOOL=0
+        echo 'export ITBPOOL=${ITBPOOL:-1}' > /etc/osg/image-init.d/01-itb.sh; \
```
Comment (brianhlin) [inline Dockerfile:129]: Any reason you're doing this instead of just setting `ENV ITBPOOL=-1` in the Dockerfile?

**Review by rynge — APPROVED** [review]
Looks good

**Review by brianhlin — APPROVED** [review]
Could you change the variable to `ITB_OSPOOL`? We may have an ITB PATh facility or something in the future! One small comment but otherwise LGTM

File: Dockerfile (hunk 1 above)
Comment (matyasselmeci) [reply Dockerfile:129]: I only want to set the default of ITBPOOL if the `ITB` build arg is set, and Docker doesn't really have `if` statements.

**Pull request merged**